x-pack/filebeat/input/cel/file_auth.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cel
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"time"
+)
+
+type fileAuthTransport struct {
+	header  string
+	prefix  string
+	path    string
+	refresh time.Duration
+	base    http.RoundTripper
+	clock   func() time.Time
+
+	mu       sync.Mutex
+	value    string
+	loadedAt time.Time
+}
+
+func newFileAuthTransport(cfg *fileAuthConfig, base http.RoundTripper) (*fileAuthTransport, error) {
+	if cfg == nil {
+		return nil, fmt.Errorf("file auth: missing configuration")
+	}
+	if base == nil {
+		base = http.DefaultTransport
+	}
+	transport := &fileAuthTransport{
+		header:  cfg.headerName(),
+		prefix:  cfg.Prefix,
+		path:    cfg.Path,
+		refresh: cfg.refreshInterval(),
+		base:    base,
+		clock:   time.Now,
+	}
+	if err := transport.initialise(); err != nil {
+		return nil, err
+	}
+	return transport, nil
+}
+
+func (t *fileAuthTransport) initialise() error {
+	t.mu.Lock()
+	defer t.mu.Unlock()
```

Comment:
Why do we need to lock this? The only goroutine that holds `t` when this is called is the goroutine that has asked for its construction and that construction is incomplete.